include/hip/hcc_detail/driver_types.h

```diff
@@ -59,22 +59,29 @@ typedef enum hipArray_Format {
 }hipArray_Format;
 
 typedef struct HIP_ARRAY_DESCRIPTOR {
-    enum hipArray_Format format;
-    unsigned int numChannels;
-    size_t width;
-    size_t height;
-    unsigned int flags;
-    size_t depth;
+  size_t Width;
+  size_t Height;
+  enum hipArray_Format Format;
+  unsigned int NumChannels;
 }HIP_ARRAY_DESCRIPTOR;
 
+typedef struct HIP_ARRAY3D_DESCRIPTOR {
+  size_t Width;
+  size_t Height;
+  size_t Depth;
+  enum hipArray_Format Format;
+  unsigned int NumChannels;
+  unsigned int Flags;
+}HIP_ARRAY3D_DESCRIPTOR;
+
 typedef struct hipArray {
     void* data;  // FIXME: generalize this
     struct hipChannelFormatDesc desc;
     unsigned int type;
     unsigned int width;
     unsigned int height;
     unsigned int depth;
-    struct HIP_ARRAY_DESCRIPTOR drvDesc;
+    struct HIP_ARRAY3D_DESCRIPTOR drvDesc;
```

Comment:
Shall we remove DESCRIPTOR from Array structure? It will help to avoid having multiple (may be mismatching) values. And it will help not to keep HIP_ARRAY3D_DESCRIPTOR for 1D/2D Arrays.